website/pages/blog/defer-support-codegen.mdx

```diff
@@ -0,0 +1,252 @@
+---
+title: Using @defer Directive with GraphQL Code Generator
+authors: aleksandra
+tags: [graphql, codegen]
+date: 2023-04-11
+description:
+  Learn how to boost GraphQL performance using the @defer directive and GraphQL Code Generator for
+  deferred fragment field resolution.
+image: /blog-assets/defer-directive/cover.png
+thumbnail: /blog-assets/defer-directive/thumbnail.png
+---
+
+Optimization is crucial for the good performance of GraphQL applications, which can grow in
+complexity and size. The `@defer` directive in GraphQL is a feature that allows fragment fields to
+be resolved separately from the rest of the query. This can be useful when dealing with large
+queries or slow-to-resolve fields.
+
+And now, we have some exciting news to share: support for `@defer` is available in GraphQL Code
+Generator! 🥳
+
+You can start using GraphQL Code Generator to generate code for queries that include deferred
+fragments, and the generated code will automatically include support for the `@defer` directive.
+
+Let's take a closer look at how the directive works and how you can use it in your GraphQL
+applications.
+
+## How Does `@defer` Work?
+
+When a query includes a deferred fragment field, the server will return a partial response with the
+non-deferred fields first, followed by the remaining fields once they have been resolved.
+
+For example, consider the following query:
+
+```graphql
+query GetUser($id: ID!) {
+  user(id: $id) {
+    id
+    name
+    ...OrdersFragment @defer
+  }
+}
+
+fragment OrdersFragment on User {
+  orders {
+    id
+    total
+  }
+}
+```
+
+Let's say you have a query that fetches information about a user, but the `orders` field takes more
+time to resolve. You can use the `@defer` directive to first return a partial response with the `id`
+and `name` and then later return the `orders` field. This can improve the performance of your
+application by reducing the amount of time it takes to load initial data.
+
+## Using `@defer` with GraphQL Codegen
+
+Once you update the GraphQL Code Generator to the latest version and start using the `@defer`
+directive in your queries, the generated code will automatically include support for the directive.
+Below, we’re going to see an example of usage.
+
+As the `@defer` directive
+[is now supported in Apollo](https://www.apollographql.com/docs/react/data/defer/), we’re going
+explore how it works with Apollo Client. Note that you need to use Apollo Client version 3.7.0 or
+higher for the defer support to work.
```

Comment:
maybe worth mentioning here in a quote that Urql is also supported and that you can read more below on how we used it in Hive?